TestFramework/Tests.Integration/Tests/UserIntegrationTests.cs

```diff
@@ -342,12 +342,62 @@ public void TestUserProperties()
 			// Assert
 			Assert.IsNotNull(renderURL);
 			Assert.IsNotEmpty(renderURL);
-			Assert.True(renderURL.StartsWith(kinveyClient.MICHostName + "oauth/auth?client_id"));
+			Assert.True(renderURL.StartsWith(kinveyClient.MICHostName + "oauth/auth?client_id=" + TestSetup.app_key, StringComparison.Ordinal));
+		}
+
+		[Test]
+		public void TestMIC_LoginWithMIC_NormalFlow_ClientID()
+		{
+			// Arrange
+			string redirectURI = "http://test.redirect";
+			User loggedInUser = null;
+
+			// Act
+			string renderURL = null;
+			string micID = "12345";
+
+			User.LoginWithMIC(redirectURI, new KinveyMICDelegate<User>
+			{
+				onSuccess = (user) => { loggedInUser = user; },
+				onError = (e) => { Console.WriteLine("TEST MIC ERROR"); },
+				onReadyToRender = (url) => { renderURL = url; }
+			}, micID);
+
+			System.Diagnostics.Debug.WriteLine("\tClientID: " + micID);
+
+			// Assert
+			Assert.IsNotNull(renderURL);
+			Assert.IsNotEmpty(renderURL);
+			Assert.True(renderURL.StartsWith(kinveyClient.MICHostName + "oauth/auth?client_id=" + TestSetup.app_key + ":" + micID, StringComparison.Ordinal));
+		}
+
+		[Test]
+		[Ignore("Placeholder - Need configured backend to run test")]
```

Comment:
@vinaygahlawat understandably these are turned off, but it'll be nice to mock up the responses and have them run soon! :)